pkg/controller/workload/kubernetes/resource/resource.go

```diff
@@ -153,12 +165,14 @@ type remoteCluster struct {
 func (c *remoteCluster) sync(ctx context.Context, ar *v1alpha1.KubernetesApplicationResource, secrets []corev1.Secret) (v1alpha1.KubernetesApplicationResourceState, error) {
 	meta.AddFinalizer(ar, finalizerName)
 
-	// Our CRD requires template to be specified, but just in case...
-	if ar.Spec.Template == nil {
-		return v1alpha1.KubernetesApplicationResourceStateFailed, errMissingTemplate
+	// We copy the resource template here so we can modify its namespace and
+	// remote controller annotations without persisting those changes back to
+	// the KubernetesApplicationResource.
```

Comment:
Nit: this comment doesn't make as much sense now that we're rendering the template rather than copying it.